Add API package and placeholder endpoints

swoldemi opened the pull request:
Let me know what you think about https://github.com/ynigoreyes/resume-engine/blob/mux-api/pkg/api/routes.go#L21-L42 and https://github.com/ynigoreyes/resume-engine/blob/mux-api/main.go#L59-L61. 
You can `curl localhost:8080/api/user/asd && curl localhost:8080/api/comment/asd && curl -X POST localhost:8080/api/comment` (:8081 if you start it with gin) to test them.

I still need to fix the gorm models.

closes #15

swoldemi committed:
Add API package

swoldemi committed:
Formalize server startup

swoldemi committed:
Add placeholder API endpoints

ynigoreyes commented on pkg/api/routes.go:
```diff
@@ -0,0 +1,42 @@
+// AddComment exposes the endpoint necessary for adding comments
+func (ro *Routes) AddComment(w http.ResponseWriter, r *http.Request) {
+	data := struct {
+		Status       int
```
~~I don't think it's good practice to have the status in the request body.~~
So people are saying it belongs in [both](https://stackoverflow.com/a/45742094) the header and the status code in the body?
I kinda like it better in the ACTUAL header though.
Try using `w.WriteHeader(http.StatusOK)` instead

ynigoreyes replied:
Yeah you should refmt the struct so that it reflects a boilerplate response body, not a struct that contains a response body

swoldemi replied:
oof, yeah idk what i was thinking 😴. Writing the header is the right way to do it 👍

swoldemi committed:
Update User struct and add Validate method

swoldemi committed:
Update Comment struct and add Validate method

  - Comment.CommenterID is a foreign key to User.ID
  - All *.ResumeID fields removed

swoldemi committed:
Implement routes.GetComment

swoldemi committed:
Implement routes.GetUser

swoldemi committed:
Implement routes.AddComment
